listings/applications/components/src/ownable.cairo

```diff
@@ -93,3 +98,159 @@ pub mod ownable_component {
         }
     }
 }
+// ANCHOR_END: ownable_component
+
+// ANCHOR: MockContract
+#[starknet::contract]
+mod MockContract {
+    use super::{IOwnable, ownable_component, ownable_component::OwnableInternalTrait};
+
+    component!(path: ownable_component, storage: ownable, event: OwnableEvent);
+
+    #[abi(embed_v0)]
+    impl OwnableImpl = ownable_component::Ownable<ContractState>;
+
+    #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        ownable: ownable_component::Storage,
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState) {
+        self.ownable._init(starknet::get_caller_address());
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        OwnableEvent: ownable_component::Event,
```

Comment:
```suggestion
        #[flat]
        OwnableEvent: ownable_component::Event,
```
@julio4 @NueloSE I [found out](https://github.com/starkware-libs/cairo/issues/5861#issuecomment-2186267348) that adding the `#[flag]` attribute above the component's event (in the contract that imports it) fixes the [event spying issues](https://github.com/NethermindEth/StarknetByExample/pull/205#issuecomment-2182249492).
We don't have to [comment out the event asserts](https://github.com/NethermindEth/StarknetByExample/pull/205#issuecomment-2186217896).

Reply:
Ignore this comment, we can include event asserts without using the `#[flat]` attribute